Lib/zipfile/__init__.py

```diff
@@ -1839,7 +1853,27 @@ def _sanitize_windows_name(cls, arcname, pathsep):
         arcname = pathsep.join(x for x in arcname if x)
         return arcname
 
-    def _extract_member(self, member, targetpath, pwd):
+    def _apply_permissions(self, member, path, mode):
+        """
+        Apply ZipFile permissions to a file on the filesystem with
+        specified PreserveMode
+        """
+        if mode == PreserveMode.NONE:
+            return
+
+        # Ignore permissions if the archive was created on Windows
+        if member.create_system == 0:
```

Comment:
I think the check should probably be more conservative than this, there is also 10 which is Windows NTFS, and 20-255 are unused and could be invalid for this use case. An allow-list would be safer.